src/main/java/com/aws/greengrass/deployment/converter/DeploymentDocumentConverter.java

```diff
@@ -215,9 +215,13 @@ private static DeploymentPackageConfiguration convertComponent(String componentN
                 .rootComponent(true) // As of now, CreateDeployment API only gives root component
                 .configurationUpdateOperation(
                         convertComponentUpdateOperation(componentUpdate.getConfigurationUpdate()));
-        builder = builder.runWith(RunWith.builder()
-                .posixUser(componentUpdate.getRunWith() == null ? null : componentUpdate.getRunWith().getPosixUser())
-                .build());
+
+        if (componentUpdate.getRunWith() != null) {
```

Comment:
why are you changing this? I just fixed it

Reply:
@leaf94 ?

Reply:
The runwith should always be set as should posix user, even if it is null

Reply:
@leaf94 changed it so I don't know what's going on. Why does it need to be always set?

Reply:
It needs to be set in order to reset the runwith back to default when the user wants.
https://github.com/aws-greengrass/aws-greengrass-nucleus/pull/965